File: app/src/main/java/io/github/amadeu01/clone2048/viewmodel/GameViewModel.java
package io.github.amadeu01.clone2048.viewmodel;

import android.databinding.ObservableArrayMap;
import android.databinding.ObservableField;

import java.util.HashMap;

import io.github.amadeu01.clone2048.model.Board;
import io.github.amadeu01.clone2048.model.Move;

/**
 * @Author Amadeu Cavalcante Filho
 * @Date 02/03/2018
 * @Email amadeu01@gmail.com
 */

public class GameViewModel implements ViewModel {
    private Board boardModel;

    public final ObservableArrayMap<String, String> tiles = new ObservableArrayMap<>();
    public final ObservableField<String> score = new ObservableField<>();
    public final ObservableField<String> winner = new ObservableField<>();

    public GameViewModel() {
        this.boardModel = new Board();
        HashMap<String, String> tileHashMap = boardModel.newGame();
        tiles.putAll(tileHashMap);
    }

    public void onMove(Move move) {

    }

    @Override
    public void onCreate() {

    }

    @Override
    public void onPause() {

    }

    @Override
    public void onResume() {

    }

    @Override
    public void onDestroy() {

    }
}

File: app/src/main/java/me/amadeucavalcante/a2048/viewmodel/GameViewModel.kt
package me.amadeucavalcante.a2048.viewmodel

import android.databinding.ObservableArrayMap
import android.databinding.ObservableField

class GameViewModel {
    val tiles = ObservableArrayMap<String, String>()
    val score = ObservableField<String>("score")
    val winner = ObservableField<String>("Something")

    init {
        tiles["00"] = "0"
        tiles["01"] = "0"
        tiles["02"] = "0"
        tiles["03"] = "0"

        tiles["10"] = "0"
        tiles["11"] = "0"
        tiles["12"] = "0"
        tiles["13"] = "0"

        tiles["20"] = "0"
        tiles["21"] = "0"
        tiles["22"] = "0"
        tiles["23"] = "0"

        tiles["30"] = "0"
        tiles["31"] = "0"
        tiles["32"] = "0"
        tiles["33"] = "0"
    }
}